src/app/core/demo-data/demo-data-initializer.service.ts

```diff
@@ -62,13 +64,35 @@ export class DemoDataInitializerService {
 
     this.localAuthService.saveUser(this.normalUser);
     this.localAuthService.saveUser(this.adminUser);
+
     await this.sessionManager.offlineLogin(this.normalUser);
+
+    // Load and parse the demo_entities.json file before publishing demo data
+    const demoDataFromJson = await this.loadDemoEntities();
+    if (demoDataFromJson?.docs) {
+      console.log("Loaded demo_entities.json:", demoDataFromJson);
+    } else {
+      console.log(
+        "No demo_entities.json found, proceeding with default demo setup.",
+      );
+    }
+
```

Comment:
To keep the different functionalities separated and easier to understand, maybe you can create a new Angular Service ("DemoDataFromFileService" or something ...) for your logic. Then we can here in the DemoDataInitializerService (or possibly in the DemoDataService) just call the loaded or the generated data service